synapse/storage/databases/main/devices.py

```diff
@@ -1748,7 +1748,8 @@ def _add_device_outbound_room_poke_txn(
                     device_id,
                     room_id,
                     stream_id,
-                    False,
+                    # We only need to calculate outbound pokes for local users
```

Comment:
Because we never need to know (indeed, _cannot_ know) the full set of hosts that a remote user shares via their rooms?

Reply:
It's more that we would *never* send out device list updates for remote users, so there is no need to calculate where to send the update to